### ISSUE TITLE
Possible register corruption

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1660]
The library [StbImageBeef](https://github.com/StbBeef/StbImageBeef) has a method called `stbi__gif_load_next` in which at one point the registers seem to get corrupted:

https://user-images.githubusercontent.com/86157825/180703400-bae1b8b9-05a4-4bc0-be32-b0f794db4018.mp4

Code:
```cs
using System;
using System.Diagnostics;
using StbImageBeef;
using System.IO;

namespace test8
{
	public static class Program
	{
		public static void Main()
		{
			let fs = scope FileStream()..Open("blue-nightsky.gif");
			let img = ImageResult.FromStream(fs).Get();
			delete img;
		}
	}
}
```

Example GIF:
![blue-nightsky](https://user-images.githubusercontent.com/86157825/180703047-21c5536e-5655-4711-b57a-b332b3c3dca4.gif)

Tested with: https://github.com/beefytech/Beef/commit/119e9189e5cc41c747b2af01abbb87966cb6b422

**Comment by bfiete** [comment]
For me it just crashes in `stbi__gif_parse_colortable`. 

The port for this library is a bit strange...

`stbi_uc  pal[256][4];` at https://github.com/nothings/stb/blob/master/stb_image.h#L6489
was ported as
`public uint8** pal = (uint8**)stbi__malloc(256 * 4 * sizeof(uint8));` at https://github.com/StbBeef/StbImageBeef/blob/master/src/StbImageBeef/src/StbImage.Generated.Gif.bf#L520

Instead of just
`public uint8[256][4] pal;`

The current code just doesn't work... I guess in your case you ended up with memory corruption instead of just a crash.







ge

**Comment by disarray2077** [comment]
>stbi_uc pal[256][4]; at https://github.com/nothings/stb/blob/master/stb_image.h#L6489
was ported as
public uint8** pal = (uint8**)stbi__malloc(256 * 4 * sizeof(uint8)); at https://github.com/StbBeef/StbImageBeef/blob/master/src/StbImageBeef/src/StbImage.Generated.Gif.bf#L520

True. I have no idea why it was ported like this...

>For me it just crashes in stbi__gif_parse_colortable.For me it just crashes in stbi__gif_parse_colortable.

Yeah, for me it also just crashes in `stbi__gif_parse_colortable`, but while trying to debug why it crashes I found out that `g` either gets nulled or changes to something else at `stbi__gif_load_next` which in turn causes the crash since `g.pal` is garbage when `stbi__gif_parse_colortable` is called.
Looking in the disassembly, `g` changes when `jne` is executed:

https://user-images.githubusercontent.com/86157825/181251666-a0f19639-d8e1-4331-b613-a57cf976e7c8.mp4

Looking at these circumstances I can't think how it could be memory corruption, for example, `Break On Change` doesn't even work in `g` since it's a register:
![image](https://user-images.githubusercontent.com/86157825/181253750-f55cc17f-d336-4cdf-8553-6a45f30ba93c.png)

**Comment by bfiete** [comment]
It was just a debug info issue- fixed at fc075610440fc8ba5cea73768fdce21ca1d8c383

Sometimes variables assigned to registers need to be saved into memory (usually volatile registers during calls) and this variable incorrectly thought it was saved into stack memory at that code location.
